sql/src/main/java/io/crate/protocols/postgres/ConnectionContext.java

```diff
@@ -634,14 +623,22 @@ void handleSimpleQuery(ByteBuf buffer, final Channel channel) {
 
         @Override
         protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Exception {
-            ByteBuf decode = decode(in);
+            ByteBuf decode = decode(in, ctx);
             if (decode != null) {
                 out.add(decode);
             }
         }
 
-        private ByteBuf decode(ByteBuf buffer) {
+        private ByteBuf decode(ByteBuf buffer, ChannelHandlerContext ctx) {
```

Comment:
Then we will have to call ctx.pipeline() for each buffer. Probably a very minor performance impact, if at all.